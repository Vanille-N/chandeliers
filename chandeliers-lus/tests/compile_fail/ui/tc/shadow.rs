//! Checking that Lustre (unlike Rust) allows shadowing a global variable
//! with a local one.
chandeliers_lus::decl! {
    const X : int = 0;
    node test() returns (X : bool);
    let X = 0; tel
}

fn main() {}
