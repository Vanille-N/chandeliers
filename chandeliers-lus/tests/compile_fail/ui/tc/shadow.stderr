error: Type mismatch between the left and right sides: Base types should be identical: expected bool, got int
 --> tests/fail/ui/tc/shadow.rs:4:9
  |
4 |     let X = 0; tel
  |         ^^^^^
  |
note: This element has type bool
 --> tests/fail/ui/tc/shadow.rs:4:9
  |
4 |     let X = 0; tel
  |         ^
note: While this element has type int
 --> tests/fail/ui/tc/shadow.rs:4:13
  |
4 |     let X = 0; tel
  |             ^
